[wip] backupccl: Experimentally add full cluster backups

pbardea opened the pull request:
This PR introduces the concept of a full cluster backup. Full cluster
backups include a subset of system tables in the backup and restore the
system tables as part of restore.

Example usage:
`BACKUP TO 'nodelocal:///full-cluster-backup'` 
and
`RESTORE FROM 'nodelocal:///full-cluster-backup'`

Release note (enterprise change): Add support for full cluster backup
and restore.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43258)
<!-- Reviewable:end -->

pbardea committed:
backupccl: Add full cluster backups and restore

This PR introduces the concept of a full cluster backup. Full cluster
backups include a subset of system tables in the backup and restore the
system tables as part of restore.

Release note (enterprise change): Add support for full cluster backup
and restore.
